frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+            </td>
+            <td
+                role="button"
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.TransportResults}:`,
+                        body: receiver.TransportResults,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.TransportResults}
+            </td>
+        </tr>
+    );
 };
 
 export const MessageReceivers = ({ receiverDetails }: MessageReceiverProps) => {
+    const modalRef = useRef<ModalRef>(null);
+    const [modalText, setModalText] = useState({ title: "", body: "" });
+    const [activeColumn, setActiveColumn] = useState("");
+    const [activeColumnSortOrder, setActiveColumnSortOrder] = useState(
+        FilterOptionsEnum.NONE
+    );
+    // Since we're splicing together both data from receiverDetails && fileUrl
+    // To simplify any proceeding logic within this component, we need to
+    // merge and normalize the data. The data currently appears like:
+    //   {
+    //     "reportId": "4b3c73df-83b1-48f9-a5a2-ce0c38662f7c",
+    //     "receivingOrg": "ignore",
+    //     "receivingOrgSvc": "HL7_NULL",
+    //     "transportResult": null,
+    //     "fileName": null,
+    //     "fileUrl": "http://azurite:10000/devstoreaccount1/reports/batch%2Fignore.HL7_NULL%2Ftx-covid-19-4b3c73df-83b1-48f9-a5a2-ce0c38662f7c-20230203182255.internal.csv",
+    //     "createdAt": "2023-02-03T18:22:55.580322",
+    //     "qualityFilters": []
+    // }
+    // And we parse fileUrl using parseFileLocation to find: folderLocation, sendingOrg and fileName. We use normalizedData to fix this. If there's no data, we fill in the string with N/A.
+
+    // Our expected output should look like the following:
+    //   NormalizedReceiverData {
+    //     Name: string;
+    //     Service: string;
+    //     Date: string;
+    //     ReportId: string;
+    //     Main: string;
+    //     Sub: string;
+    //     FileName: string;
+    //     TransportResults: string;
+    // }
+    const normalizedData = useMemo(
+        () =>
+            receiverDetails.map(
+                (receiverItem: ReceiverData): NormalizedReceiverData => {
+                    const formattedData = Object.keys(ColumnDataEnum).reduce(
+                        (accumulator: any, currentValue: string) => {
+                            accumulator[currentValue] = "N/A";
+                            return accumulator;
+                        },
+                        {}
+                    );
+                    for (const key of Object.keys(ColumnDataEnum)) {
+                        const columnTitle =
+                            ColumnDataEnum[key as keyof typeof ColumnDataEnum];
+                        switch (true) {
```

Comment:
I feel like `if-else`s are clearer than `switch (true)` 🤔

Reply:
I can actually optimize this `switch` statement to it's perfect form: 

```
switch (columnTitle) {
  case ColumnDataEnum.Name:
  case ColumnDataEnum.Name:
// etc
}
```

which I think is actually easier to read than a ton of if-else statements. In addition, from what I've read online, `switch` statements will be read faster in that the compiler will "go to" the `true` condition immediately instead of with if-else ladder, it would have to descend one-at-a-time before getting to the "truthy" value.

Reply:
Fantastic -- definitely would be best to use your proposed changed here :)

Regarding whether or not it's best to use `switch` versus `if-else`...this is a hotly debated topic 😆 Compiled languages have more definitive answers for this, but I'd say the performance benefits are negligible in JavaScript -- especially since the interpretation/underlying implementation differs between browsers -- so we should opt for human readability whenever.  

[Here](https://jsben.ch/5w6eT)'s a quick comparison if you're curious!  (Also RIP jsperf?)